### PR TITLE
Populate the Index

### PR DESCRIPTION
**Pull request by encukou**
This makes the Index page more useful to people searching for specific problems.

### CODE REVIEW

**Comment by hroncok** [comment]
Will review this today.

File: source/builtins.rst [hunk 1]
```diff
@@ -39,6 +42,9 @@ The recommended fixer will add the future import and rewrite all uses
 of ``print``.
 
 
+.. index:: input
+.. index:: NameError; raw_input
```
Comment (hroncok) [inline source/builtins.rst]: Should this also have:

```rst
.. index:: raw_input
```

To be under _R_?

File: source/classes.rst [hunk 2]
```diff
@@ -4,6 +4,11 @@ Classes
 Python 3 drops support for “old-style” classes, and introduces dedicated syntax
 for metaclasses. Read on for details.
 
+
+.. index:: object; as base class
```
Comment (hroncok) [inline source/classes.rst]: Why "as base class"?

File: source/builtins.rst (hunk 1 above)
Comment (hroncok) [reply source/builtins.rst]: There are other similar instances of this. Let's keep the discussion here, but I will comment on them nevertheless.

File: source/comparisons.rst [hunk 3]
```diff
@@ -34,6 +39,9 @@ implement a key function to fully describe how disparate types
 should be ordered.
 
 
+.. index:: __eq__, __ne__, __lt__, __le__, __gt__, __ge__
+.. index:: TypeError; __cmp__
```
Comment (hroncok) [inline source/comparisons.rst]: Bare `__cmp__` as well?

File: source/comparisons.rst [hunk 4]
```diff
@@ -182,6 +193,9 @@ we recommend adding the full, documented function to your project's utility
 library.
 
 
+.. index:: cmp; argument of sort()
+.. index:: TypeError; key function
```
Comment (hroncok) [inline source/comparisons.rst:197]: Bare `key function` as well?

File: source/dicts.rst [hunk 5]
```diff
@@ -3,6 +3,9 @@ Dictionaries
 
 There are three most significant changes related to dictionaries in Python 3.
 
+
+.. index:: AttributeError; has_key()
```
Comment (hroncok) [inline source/dicts.rst]: Bare `has_key()`?

File: source/numbers.rst [hunk 6]
```diff
@@ -80,6 +85,8 @@ This can be done with a simple alias::
         __truediv__ = __div__
 
 
+.. index:: TypeError; int & long
```
Comment (hroncok) [inline source/numbers.rst]: Bare `long`?

**Comment by hroncok** [comment]
I've asked several questions about some possible bare terms. Also, not sure about the "as base class". Other than that, LGTM.

**Review by hroncok — APPROVED** [review]
Regardless the comments, this is a significant improvement.

File: source/builtins.rst (hunk 1 above)
Comment (encukou) [reply source/builtins.rst]: +1, I'll change these. Thanks for catching them!

File: source/comparisons.rst (hunk 4 above)
Comment (encukou) [reply source/comparisons.rst:197]: Here I think it's unnecessary: "key function" comes from the error message and I don't think it's relevant on its own.

File: source/classes.rst (hunk 2 above)
Comment (encukou) [reply source/classes.rst]: I think it's relevant here – `object` can mean many things, even in the context of porting. I wanted to be clear that this is about `object` added as a base class as part of porting,

Comment (hroncok) [reply source/classes.rst]: Oh, I see. What about: `object; object as base class`?

File: source/comparisons.rst (hunk 4 above)
Comment (hroncok) [reply source/comparisons.rst:197]: ok

File: source/classes.rst (hunk 2 above)
Comment (encukou) [reply source/classes.rst]: I don't think that's necessary? I think it's assumed that such sub-entries in an index relate to the parent entry. See “raise; new syntax” for example.
But if you think it's clearer I can add it.

Comment (hroncok) [reply source/classes.rst]: I don't have strong opinion, I just couldn't understand the text as is.

Comment (encukou) [reply source/classes.rst]: OK, I've changed it

**Pull request merged**